packages/@lwc/module-resolver/src/index.ts

```diff
@@ -6,154 +6,113 @@
-    return { entry, moduleSpecifier, moduleName, moduleNamespace };
+export interface LwcConfig {
+    modules: ModuleSpecifier[];
 }
 
-function loadLwcConfig(modulePath) {
-    const packageJsonPath = path.join(modulePath, 'package.json');
-    const lwcConfigPath = path.join(modulePath, LWC_CONFIG_FILE);
-    let config;
-    try {
-        config = JSON.parse(fs.readFileSync(lwcConfigPath, 'utf8'));
-    } catch (ignore) {
-        // ignore
-    }
-    if (!config) {
-        try {
-            config = JSON.parse(fs.readFileSync(packageJsonPath, 'utf8')).lwc;
-        } catch (ignore) {
-            // ignore
+export function resolveModulesFromDir(modulesDir: string): RegistryEntry[] {
+    const namespaces = fs.readdirSync(modulesDir);
+    const resolvedModules: RegistryEntry[] = [];
+    namespaces.forEach(ns => {
+        const namespacedModuleDir = path.join(modulesDir, ns);
+        if (fs.lstatSync(namespacedModuleDir).isDirectory()) {
+            const modules = fs.readdirSync(namespacedModuleDir);
+            modules.forEach(moduleName => {
+                const moduleDir = path.join(namespacedModuleDir, moduleName);
+                const entry = getModuleEntry(moduleDir, moduleName);
+                if (entry) {
+                    const specifier = `${ns}/${moduleName}`;
+                    resolvedModules.push({ entry, specifier });
+                }
+            });
         }
-    }
-    return config;
-}
+    });
 
-export function resolveModulesInDir(absPath: string): { [name: string]: RegistryEntry } {
-    return glob
-        .sync<FlatEntry>(MODULE_ENTRY_PATTERN, {
-            cwd: absPath,
-            transform: entry =>
-                typeof entry === 'string' ? { path: entry } : { path: entry.path },
-        })
-        .reduce((mappings, { path: file }) => {
-            const ext = path.extname(file);
-            const fileName = path.basename(file, ext);
-            const rootDir = path.dirname(file);
-            const rootParts = rootDir.split('/'); // the glob library normalizes paths to forward slashes only - https://github.com/isaacs/node-glob#windows
-            const entry = path.join(absPath, file);
-
-            const dirModuleName = rootParts.pop();
-            const dirModuleNamespace = rootParts.pop();
-            if (dirModuleNamespace && dirModuleName === fileName) {
-                const registryNode = createRegistryEntry(
-                    entry,
-                    `${dirModuleNamespace}/${fileName}`,
-                    fileName,
-                    dirModuleNamespace.toLowerCase()
-                );
-                mappings[registryNode.moduleSpecifier] = registryNode;
-            }
-
-            return mappings;
-        }, {});
+    return resolvedModules;
 }
 
-function hasModuleBeenVisited(module, visited) {
-    return visited.has(module);
-}
+export function resolveModulesFromNpm(packageName: string): RegistryEntry[] {
```

Comment:
If we were to do a major version bump, I think we should also remove this API. The `resolveModules` should be enough IMO.